src/index.ts

```diff
@@ -352,25 +366,63 @@ export function startAnalysis<
 
                     inProgressTrafficCollectionName = name ?? new Date().toISOString();
 
-                    const flowsOutputPath = temporaryFile();
+                    const flowsOutputPath = temporaryFile({ extension: 'har' });
+
                     mitmproxyState = {
-                        proc: execa('mitmdump', ['-w', flowsOutputPath]),
+                        proc: execa(
+                            'mitmdump',
+                            [
+                                ...Object.entries(mitmproxyAddons).map(([addonName, addonPath]) => {
+                                    if (!existsSync(addonPath))
+                                        throw new Error(
+                                            `No "${addonName}" addon for mitmproxy found at "${addonPath}".${
+                                                addonName === 'ipcEvents' || addonName === 'harDump'
+                                                    ? ' The mitmproxy capability requires that this addons is present.'
+                                                    : ''
+                                            }`
+                                        );
+                                    return `-s ${addonPath}`;
+                                }),
+                                `--set hardump=${flowsOutputPath}`,
+                            ],
+                            {
+                                stdio: ['pipe', 'pipe', 'pipe', 'ipc'],
+                                env: { ...process.env, IPC_PIPE_FD: '3' },
```

Comment:
Can you just hardcode that? :o

Reply:
Oh wait, I think I understand. These aren't global but per-process and mitmproxy "inherits" (?) this when execa `fork()`s. And the number corresponds to the index in the `stdio` array?

Reply:
Exactly